server/src/typeChecker/typeChecker.ts

```diff
@@ -182,7 +197,11 @@ export class TypeChecker
     const { tracker } = decl.identifier;
 
     if (this.isBasicTracker(tracker)) {
-      tracker.declareType(result.type.assignmentType());
+      if (decl.typeHint && TypeHint.get(decl.typeHint.toString())) {
+          tracker.declareType(TypeHint.get(decl.typeHint.toString()) || result.type.assignmentType());
```

Comment:
I think a future PR we could try to report an warning if the typehint isn't assignable to the inferred type. Maybe something like this
```
local x is "some string" // #type: scalar 
                         //  ^--- could probably indicate that scalar isn't assignable to string

local y is ship:parts[0] // #type: engine
                         // ^-- engine is a part so this should be ok
```

The common case will be structure since that when the type checker gives up which anything should be assignable to it